chattore/src/main/kotlin/feature/Alias.kt

```diff
@@ -0,0 +1,128 @@
+package chattore.feature
+
+import chattore.ChattORE
+import chattore.Feature
+import chattore.render
+import chattore.toComponent
+import com.velocitypowered.api.command.SimpleCommand
+import com.velocitypowered.api.event.Subscribe
+import com.velocitypowered.api.event.connection.PluginMessageEvent
+import com.velocitypowered.api.proxy.Player
+import com.velocitypowered.api.proxy.messages.MinecraftChannelIdentifier
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.cbor.Cbor
+import kotlinx.serialization.encodeToByteArray
+import kotlinx.serialization.json.Json
+import org.openredstone.chattore.common.AliasMessage
+import org.openredstone.chattore.common.ALIAS_CHANNEL
+import org.slf4j.Logger
+
+private val IDENTIFIER: MinecraftChannelIdentifier = MinecraftChannelIdentifier.from(ALIAS_CHANNEL)
+
+private val argumentsRegex = Regex("\\\$(args|[0-9])")
+
+@Serializable
+data class AliasConfig(
+    val alias: String = "alias",
+    val commands: List<String> = listOf("some", "commands"),
+)
+
+fun createAliasFeature(
+    plugin: ChattORE
+): Feature {
+    plugin.proxy.channelRegistrar.register(IDENTIFIER)
+    fun String.toCommandAliasMeta() = plugin.proxy.commandManager.metaBuilder(this)
+        .plugin(plugin)
+        .build()
+    val resourcePath = "aliases.json"
+    val resourceBytes = plugin.loadResource(resourcePath)
+    val aliasesJson = resourceBytes.decodeToString()
+    val aliases = Json.decodeFromString<List<AliasConfig>>(aliasesJson)
+    aliases.forEach { (alias, commands) ->
+        val meta = alias.toCommandAliasMeta()
+        val aliasCommand = AliasCommand(plugin, alias, commands)
+        plugin.proxy.commandManager.register(meta, aliasCommand)
+    }
+    plugin.logger.info("Loaded ${aliases.size} aliases")
+    return Feature(listeners = listOf(ClientAliasListener(plugin.logger)))
+}
+
+class AliasCommand(
+    private val plugin: ChattORE,
+    private val base: String,
+    private val commands: List<String>
+) : SimpleCommand {
+
+    private val requiredArgs = commands.maxOfOrNull { argumentsRegex.findAll(it).toList().size } ?: 0
```

Comment:
Idk what kinda weird alias we'd have to have in order for this to be a problem, but it would be more robust if it finds the maximum of the used `$n`s of all commands :thinking:

Reply:
Ah, true yeah.